client/src/components/Users/Auth/SignUpUserAndNonprofit/validation-rules.ts

```diff
@@ -2,11 +2,12 @@ import { DeepRequired, FieldErrorsImpl } from 'react-hook-form';
 import { FormData } from './FormData';
 import * as Yup from 'yup';
 import { classifications } from './Classifications';
+import { US_STATE_ABBREVIATIONS } from '../../../../configs';
 
 export const validationSchema = Yup.object().shape({
   doing_business_as: Yup.string().required('Required'),
   city: Yup.string().required('Required'),
-  state: Yup.string().required('Required').min(2).max(2),
+  state: Yup.string().required('Required').oneOf(US_STATE_ABBREVIATIONS),
```

Comment:
looks like `.oneOf` can take an error message as the 2nd arg as well
https://github.com/jquense/yup#schemaoneofarrayofvalues-arrayany-message-string--function-schema-alias-equals
i'd vote to pass in a string like `"must be a valid US state abbreviation"` or something